README.MD: Updated Formatting & Grammar

Galagyy opened the pull request:
Updated the README.md to have better Grammar and Formatting making it easier to read and work with.

Changes:
- Added a Table of Contents.
- Fixed grammar.
- Added anchors to sections.
- Named images/hyperlinks relative to their game.

Galagyy commented:
If you want to view the changed README, you can find it [here](https://github.com/Galagyy/thprac/tree/master).

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+* [Credits](#credits)
+* [FAQ](#faq)
+
+## <a name="downloading"></a> **Downloading**
```
I put this at the very very top because this GitHub page is essentially thprac's website. The problem is that people don't know how to use GitHub, so I want a direct download button as high up on the page as possible.

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+You can use this tool in a variety of ways; the main ways to do this are **placing the executable in the same folder as your game**, **running the executable after running your game**, or **using the thprac launcher**.
+
+### **Using the folder**
+**This method will NOT work with Steam games.** You can run the thprac tool by placing the `thprac.exe` file in the same folder as your game. This method will also automatically detect vpatch. You must ensure that the game is updated to the latest version and follows one of the following naming schemes. This method also works with vpatch.
```
please refer to thprac like it's just a noun, and don't call it "the thprac tool"

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+- 東方紅魔郷.exe (Embodiment of Scarlet Devil)
+- alcostg.exe (Uwabami Breakers)
+
+### **Running the Executable**
```
this should probably be called something like "attaching thprac after running the game"

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+## <a name="usage"></a> **Usage**
+You can use this tool in a variety of ways; the main ways to do this are **placing the executable in the same folder as your game**, **running the executable after running your game**, or **using the thprac launcher**.
+
+### **Using the folder**
```
This should probably be called something like "with thprac in the game folder"

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
-## Compatibility
-thprac officially supports every version of Windows starting with **Windows Vista**. Windows XP may work using [One-Core-API](https://github.com/Skulltrail192/One-Core-API-Binaries), but this is NOT being actively tested.
+## <a name="compatability"></a> **Compatability**
+thprac is compatible with all Windows versions proceeding and including Windows Vista. You may attempt to apply thprac on Windows XP using the [One-Core-API](https://github.com/Skulltrail192/One-Core-API-Binaries) but this is not guaranteed to work.
```
I like what I previously wrote here better. For reference, that would be "thprac officially supports every version of Windows starting with Windows Vista. Windows XP may work using [One-Core-API](https://github.com/Skulltrail192/One-Core-API-Binaries), but this is NOT being actively tested."

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+thprac is also compatible with **Wine** and the Steam Deck. Do note that the backspace menu is tailored to work well with the Steam Deck, although this is untested.
+
+## <a name="features"></a> **Features**
+thprac features an enhanced practice mode that supports all main games along with Great Fairy Wars and Uwabami Breakers; do note that thprac **does not support Phantasmagoria of Flower View**.
```
thprac does support PoFV (and both versions of UDoALG, though support for those games works differently). I consider TH09 Tools and TH19 Tools to basically be the enchanced practice mode for those games. Actually, looking one line further down, I understand why you wrote that. You wrote that because it says "Does not support PoFV" in the line you removed.

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
-## Backspace menu
-In any supported game (except for TH09) you can press BACKSPACE to access the Backspace menu. It will show you options that can be activated with the F keys.
+## <a name="backspace-menu"></a> **Backspace Menu**
+In any supported game (**except Touhou 9**), you can press the **backspace** key to enter the backspace menu; the backspace menu will show you a variety of options that can be activated using the function (F) keys.
```
except Touhou 9 and Touhou 19

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+In any supported game (**except Touhou 9**), you can press the **backspace** key to enter the backspace menu; the backspace menu will show you a variety of options that can be activated using the function (F) keys.
 
-NOTE for Steam Deck users: these options are also **clickable**. This means that you can use your left thumb to tap the options to toggle them. Therefore, you only need to map backspace in Steam Input.
+Do note that on the **Steam Deck** the options are clickable; you can use your left thumb to tap the options to toggle them. As a result, you only need to map the backspace key in Steam Input.
```
These options are always clickable (you can click on them with the mouse when running on a desktop computer). I added that so that these options could be tapped on if you are using a Steam Deck. To say "that on the Steam Deck the options are clickable" is technically inaccurate though.

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+![Unconnected Marketeers](https://user-images.githubusercontent.com/23106652/174435006-e906d30d-0ef5-4930-ae57-1f0919beb5af.png)
+
+## <a name="statistics"></a> **Statistics (WIP)**
+If you are playing Touhou 13, you can press the "L" key to access a more advanced HUD.
```
No longer a thing (though I understand why you kept that)

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
-### TH185 Force Wave
-![grafik](https://github.com/touhouworldcup/thprac/assets/23106652/02c55e5e-5c89-462f-beea-9ab07cbb1051)
+## <a name="game-specific-features"></a> **Game-specific Features**
+Some games such as **100th Black Market** and **Phantasmagoria of Flower View** have game specific features to make practicing easier.
```
No need for that line

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
 ### ESC + R fix for Violet Detector, Wily Beast and Weakest Creature and Unconnected Marketeers
 
-# Credits (May 27th 2022 and after)
+## <a name="credits"></a> **Credits (post May 27th, 2022)**
```
don't forget to credit yourself for revising the readme :)

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+
+However, EoSD now supports this functionality, albeit with quirks. 
+
+Note: replays saved mid-game do not return to the title screen after the save point.
```
The note "albeit with quirks" is not really needed when the one and only quirk is described right below

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+
+### How do I switch languages?
+- The launcher automatically selects its language based on your Windows settings.  
+- To change the language in-game, use the shortcuts:  
```
Language 1 is Japanese, language 2 is Chinese, language 3 is English

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+Visit the [Issues tab on GitHub](https://github.com/touhouworldcup/thprac/issues) to report bugs or suggest improvements.
 
-# How to build:
-A Visual Studio solution is bundled that uses the Visual Studio 2022 build tools (v143) for building. However, thprac will compile perfectly fine with Visual Studio 2019 build tools (v142) as well. Your builds will support Windows XP perfectly fine, v141_xp build tools are not needed.
```
thanks for removing that line. That line has been out of date for a while (I switched to v143 so I could use C++20)

32th-System commented on README.md:
```diff
@@ -1,128 +1,207 @@
+![100th Black Market](https://github.com/touhouworldcup/thprac/assets/23106652/02c55e5e-5c89-462f-beea-9ab07cbb1051)
 
-If thprac is enabled ("Mode" is set to "Custom"), this popup will appear whenever the game is about to randomly choose a wave.
+If thprac is enabled using "Custom" mode, the popup above will appear when the game is about to choose a wave.
```
The reason I phrased it as "Mode" is set to "Custom" is because in this dropdown that's labelled "Mode" the option that the user has to select is called "Custom"
![grafik](https://github.com/user-attachments/assets/5b54bbbe-dc46-45e8-bbc8-e93de98a6e14)

32th-System reviewed (changes requested):
Thanks for doing this, this rewrite was long overdue, as evident by the fact that things that are no longer true made it into the new README, because they were still in the old README.

I just want to let you know that I am in no way trying to blame you for putting misinformation in the new README, when you got that information... from the old README.

I'll try to contact TNT569 to get this translated into Chinese.

Galagyy commented:
Hello, thank you for the prompt response, I'll go ahead and address the changes you prompted.

Galagyy commented:
Just an FYI, I'll place each of the concerns in their own commit to make reviewing them much easier, thank you.

Galagyy commented:
I've gone ahead and added your suggestions in this commit, thank you for the review!

I was a little bit confused with this comment, but I took the main idea from it and removed the message stating no support for PoFV:
> "thprac does support PoFV (and both versions of UDoALG, though support for those games works differently). I consider TH09 Tools and TH19 Tools to basically be the enchanced practice mode for those games. Actually, looking one line further down, I understand why you wrote that. You wrote that because it says "Does not support PoFV" in the line you removed."

https://github.com/touhouworldcup/thprac/pull/202/commits/b059e71d7a7ee6e0102d3ce01f4c11433457dede

32th-System commented:
Nice, the download link is moved further up. I would still have the link just be named "Download" though, and the other link would be "Download Beta". And I guess you could also add a note below that says "Note: beta versions will auto-update to the next stable version, but NOT to other beta versions"

Galagyy commented:
Hello, thanks for the response, I've changed it to say "Download" and "Download Beta" and added the note to both sections:

![image](https://github.com/user-attachments/assets/e3360c7d-ee62-4454-a64a-52a60462027d)
